core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java

```diff
@@ -181,42 +184,92 @@ Map<String,String> modifyProperties(Consumer<Map<String,String>> mapMutator)
    *
    * @return a list of locations in <code>hostname:port</code> form.
    * @since 2.1.0
+   * @deprecated see {@link #getServers(ServerTypeName)}
    */
+  @Deprecated(since = "4.0.0")
   List<String> getManagerLocations();
 
   /**
    * Returns the locations of the active compactors
    *
    * @return A set of currently active compactors.
    * @since 2.1.4
+   * @deprecated see {@link #getServers(ServerTypeName)}
    */
+  @Deprecated(since = "4.0.0")
   Set<String> getCompactors();
 
   /**
    * Returns the locations of the active scan servers
    *
    * @return A set of currently active scan servers.
+   * @deprecated see {@link #getServers(ServerTypeName)}
    * @since 2.1.0
    */
+  @Deprecated(since = "4.0.0")
   Set<String> getScanServers();
 
   /**
    * List the currently active tablet servers participating in the accumulo instance
    *
    * @return A list of currently active tablet servers.
+   * @deprecated see {@link #getServers(ServerTypeName)}
    */
+  @Deprecated(since = "4.0.0")
   List<String> getTabletServers();
 
+  /**
+   * Resolve the server of the given type and address to a ServerId
+   *
+   * @param type type of server
+   * @param resourceGroup group of server, can be null
+   * @param host host name, cannot be null
+   * @param port host port
+   * @return ServerId if found, else null
+   * @since 4.0.0
+   */
+  ServerId getServer(ServerTypeName type, String resourceGroup, String host, int port);
+
+  /**
+   * Returns all servers of the given types
+   *
+   * @return set of servers of the supplied type
+   * @since 4.0.0
+   */
+  Set<ServerId> getServers(ServerTypeName type);
+
+  /**
+   * Returns the servers of a given type that match the given criteria
+   *
+   * @return set of servers of the supplied type matching the supplied test
+   * @since 4.0.0
+   */
+  Set<ServerId> getServers(ServerTypeName type, Predicate<ServerId> test);
```

Comment:
Opened #4943, thinking in a follow on PR when this and #4943 are merged that we can have a method like

```java
Set<ServerId> getServers(ServerTypeName type, Predicate<String> resourceGroupPredicate, BiPredicate<String, Integer> hostPortPredicate)
```

Had planned on completely reviewing this, but got distracted by #4943.  Plan on finishing review this tomorrow.

Reply:
I just reviewed #4943. I think we should merge that in before this is merged.

Reply:
#4943 and #4944 have been merged into `main`, and `main` merged into this branch. I should have all of your changes.